server/jobs/moss.py

```diff
@@ -57,6 +59,20 @@ def submit_to_moss(moss_id=None, file_regex=".*", assignment_id=None, language=N
         match_pattern = re.compile(file_regex)
         ignored_files = set()
 
+        template_files = []
+        for template in assign.files:
+            dest = "{}/{}".format(tmp_dir, template)
+            with open(dest, 'w') as f:
+                f.write(assign.files[template])
+            template_files.append(template)
```

Comment:
template_files is just list(assign.files). You could pass assign.files to the ''.join below directly